en/drafts/originals/analyzing-multilingual-text-nltk-spacy-stanza.md

```diff
@@ -852,11 +821,11 @@ word: .	upos: PUNCT
 
 ### Lemmatization
```

Comment:
I'm happy to add in the lemmatization of the Russian sentence with Stanza, and will await further comment from @lachapot before making changes.

Reply:
Thanks for all the great work you’ve both done on this! Sorry for the delay in getting back to you @ian-nai  and @charlottejmc. 

Adding in the lemmatization of the Russian sentence with Stanza would be great. 

Maybe we could also add a little more explanation in the sentence: “spaCy does not have a single multilingual lemmatization corpus, so we’ll have to split our multilingual sentence into its component parts again, then run separate models on the Russian and French text.” For example, something like: “As with POS tagging above, it is only possible to lemmatize monolingual text with spaCy because spaCy does not have a multilingual model for lemmatization. So in order to lemmatize our text, we’ll have to split our multilingual sentence into its component parts again, then run separate models on the split Russian and French text.”

After that, perhaps we could add a few sentences before the conclusion in which we pick out examples in the output to explain what lemmatization has done and how it might useful? For example, pointing out that “vois” has been been lemmatized to “voir”, and if we were interested in identifying instances of the verb to see or the theme of seeing and vision in the text, lemmatization is necessary so that we can reliably identify all instances regardless of inflection or variation in the language. This is particularly useful if we then want to use additional methods for further analysis, for example word frequency counts or different kinds of counting methods to analyze the frequency of words in a text. (Or something similar, just to explain a little more what the outputs mean and how this can be useful). 

This could also then be a good transition into the conclusion where the key elements of the lesson are summed up alongside pointers to how they may be applied in further analysis.

Those would be the suggestions I have for fleshing out the lemmatization section and smoothing the transition into the conclusion. What do you think? Hope this helps!

Reply:
Thanks @ian-nai! This looks great to me. Perhaps we could just add a small clarification in the introductory sentence, just to be clear which text/sentences we're working with (e.g. Now we do the same with Stanza _and lemmatize the Russian sentence, French sentence, and multilingual sentence_.)? Then this mirrors the introductory sentence to spaCy above (that specifies that we’re lemmatizing the sentences grouped together by language) and helps with flow. But otherwise I think this looks good!

Reply:
Thanks, @lachapot! That clarification to the introductory sentence sounds good to me!

Reply:
Thank you @ian-nai and @lachapot! 

I've added the changes to the Lemmatization section, with some minor copyedits (as you can see in [this commit]()). 

I am still a little bit confused by the lemmatization with spaCy section: it seems to me as though we are only lemmatizing the multilingual sentence (split into the Russian parts first, then the French)? I've updated the introductory sentence to say:

 'spaCy does not have a single multilingual lemmatization corpus, so we'll have to split our multilingual sentence into its component parts again, then run separate models on the Russian and French sections of the sentence.' 

Perhaps we could also insert an additional sentence just to quickly explain why we are only working on the multilingual sentence here (even if it's just a matter of space/simplicity).

Let me know what you think. I think the section is already much improved, and almost ready to be merged!

Reply:
P.S. @ian-nai, you can still edit your file directly by navigating to the [files changed](https://github.com/programminghistorian/ph-submissions/pull/631/files) tab at the top of this page, then clicking the three dots and _Edit File_ like this:

  
![image](https://github.com/user-attachments/assets/08f44da8-d755-4ea8-bede-41f08404ea78)

Reply:
Thanks, @charlottejmc! Is this simple explanatory sentence okay? "To keep things simple, we'll only use the multilingual sentence as an example to demonstrate lemmatization."

The Edit File link is disabled for me when I navigate to the files changed tab, unfortunately; I'm able to comment and approve changes, but not make direct edits.

Reply:
Thanks @ian-nai. I worry the phrasing is a bit misleading, as you are still using the French and Russian sentences to demonstrate lemmatization with Stanzy (below). It's also slightly contradictory to 'keep things simple' with the multilingual sentence, yet go on to say that spaCy doesn't actually support multilingual lemmatization! I've modified it to say:

'To keep things short, we'll use only the multilingual sentence as an example to demonstrate lemmatization with spaCy. However, spaCy does not have a single multilingual lemmatization corpus, so [...]'

Thank you very much for your patience as we worked through these final knots! I'll now merge the PR, and I'll be moving on to the typesetting phase of Phase 6.